src/Persistence_matrix/include/gudhi/Persistence_matrix/base_matrix_with_column_compression.h

```diff
@@ -344,11 +350,10 @@ class Base_matrix_with_column_compression : protected Master_matrix::Matrix_row_
   index nextColumnIndex_;                             /**< Next unused column index. */
   Column_settings* colSettings_;                      /**< Cell factory. */
   /**
-   * @brief Column factory.
-   * @warning As the member is static, they can eventually be problems if the matrix is duplicated in several threads.
-   * If this become necessary, the static should be removed in the future.
+   * @brief Column factory. Has to be a pointer as Simple_object_pool is not swappable, so their adresses have to be
+   * exchanged instead.
    */
-  inline static Simple_object_pool<Column_type> columnPool_;
+  Simple_object_pool<Column_type>* columnPool_;
```

Comment:
```suggestion
  std::unique_ptr<Simple_object_pool<Column_type>> columnPool_;
```
then you can remove delete from the destructor.

Reply:
Is there any other advantage other than to remove the `delete`?

Reply:
It has a move constructor that does the right thing, which helps avoid explicitly defining one in classes that contain it.
If built with `make_unique`, it removes a potential memory leak in case of exceptions.
Mostly, it clearly documents ownership.